finalitygadget/finalitygadget.go

```diff
@@ -413,47 +460,75 @@ func (fg *FinalityGadget) ProcessBlocks(ctx context.Context) error {
 				return fmt.Errorf("error querying BTC staking activation timestamp: %w", err)
 			}
 
-			// only process blocks after the btc staking is activated
+			// throw error if btc staking activated before the first block was finalized (see startup order above)
+			if latestFinalizedHeight == 0 && latestFinalizedBlockTime < btcStakingActivatedTimestamp {
```

Comment:
shouldn't the logic be 

btcStakingActivatedTimestamp < latestFinalizedBlockTime

Reply:
good spot - updated!